openvdb_points/unittest/TestPointDataLeaf.cc

```diff
@@ -862,6 +919,58 @@ TestPointDataLeaf::testIO()
 }
 
 
+void
+TestPointDataLeaf::testSwap()
+{
+    using namespace openvdb::tools;
+
+    // Define and register some common attribute types
+
+    typedef TypedAttributeArray<float>    AttributeS;
+    typedef TypedAttributeArray<int32_t>  AttributeI;
+
+    AttributeS::registerType();
+    AttributeI::registerType();
+
+    // create a descriptor
+
+    typedef AttributeSet::Descriptor Descriptor;
+
+    Descriptor::Inserter names;
+    names.add("density", AttributeS::attributeType());
+    names.add("id", AttributeI::attributeType());
+
+    Descriptor::Ptr descrA = Descriptor::create(names.vec);
+
+    // create a leaf and initialize attributes using this descriptor
+
+    const size_t initialArrayLength = 100;
+    LeafType leaf(openvdb::Coord(0, 0, 0));
+    leaf.initializeAttributes(descrA, /*arrayLength=*/initialArrayLength);
+
+    // swap out the underlying attribute set with a new attribute set with a matching
+    // descriptor
+
+    CPPUNIT_ASSERT_EQUAL(initialArrayLength, leaf.attributeSet().get("density")->size());
+    CPPUNIT_ASSERT_EQUAL(initialArrayLength, leaf.attributeSet().get("id")->size());
+
+    const size_t newArrayLength = initialArrayLength / 2;
+    leaf.swap(new AttributeSet(descrA, /*arrayLength*/newArrayLength));
+
+    CPPUNIT_ASSERT_EQUAL(newArrayLength, leaf.attributeSet().get("density")->size());
+    CPPUNIT_ASSERT_EQUAL(newArrayLength, leaf.attributeSet().get("id")->size());
+
+    // ensure we refuse to swap when the descriptors do not match
+
+    names.add("extra", AttributeS::attributeType());
+    Descriptor::Ptr descrB = Descriptor::create(names.vec);
+    AttributeSet* attrbSet = new AttributeSet(descrB, newArrayLength);
```

Comment:
Maybe it's just me, but I find mis-spelled words make it harder to follow code, which includes dropping random vowels (Nick knows all about my aversion to this). Abbreviations are tolerable, though I often still prefer explicit longer names. "attribSet" or "attributeSet" please. :)